container/libcontainer/compatibility.go

```diff
@@ -15,354 +15,25 @@
-			MacAddress:        net.MacAddress,
-			Address:           net.Address,
-			Gateway:           net.Gateway,
-			IPv6Address:       net.IPv6Address,
-			IPv6Gateway:       net.IPv6Gateway,
-			HostInterfaceName: state.NetworkState.VethHost,
-		}
-		result.Networks = append(result.Networks, n)
-	}
-	result.Routes = config.Routes
-	if config.Cgroups != nil {
-		result.Cgroups = config.Cgroups
-	}
-
-	return &result, nil
-}
-
-func convertOldConfigToNew(config v1Config) *configs.Config {
-	var (
-		result configs.Config
-		old    *v1Cgroup = config.Cgroup
-	)
-	result.Rootfs = config.Config.Rootfs
-	result.Hostname = config.Config.Hostname
-	result.Namespaces = config.Config.Namespaces
-	result.Capabilities = config.Config.Capabilities
-	result.Networks = config.Config.Networks
-	result.Routes = config.Config.Routes
-
-	var newCgroup = &configs.Cgroup{
-		Name:   old.Name,
-		Parent: old.Parent,
-		Resources: &configs.Resources{
-			AllowAllDevices:   old.Resources.AllowAllDevices,
-			AllowedDevices:    old.Resources.AllowedDevices,
-			DeniedDevices:     old.Resources.DeniedDevices,
-			Memory:            old.Resources.Memory,
-			MemoryReservation: old.Resources.MemoryReservation,
-			MemorySwap:        old.Resources.MemorySwap,
-			KernelMemory:      old.Resources.KernelMemory,
-			CpuShares:         old.Resources.CpuShares,
-			CpuQuota:          old.Resources.CpuQuota,
-			CpuPeriod:         old.Resources.CpuPeriod,
-			CpuRtRuntime:      old.Resources.CpuRtRuntime,
-			CpuRtPeriod:       old.Resources.CpuRtPeriod,
-			CpusetCpus:        old.Resources.CpusetCpus,
-			CpusetMems:        old.Resources.CpusetMems,
-			BlkioWeight:       old.Resources.BlkioWeight,
-			BlkioLeafWeight:   old.Resources.BlkioLeafWeight,
-			Freezer:           old.Resources.Freezer,
-			HugetlbLimit:      old.Resources.HugetlbLimit,
-			OomKillDisable:    old.Resources.OomKillDisable,
-			MemorySwappiness:  old.Resources.MemorySwappiness,
-			NetPrioIfpriomap:  old.Resources.NetPrioIfpriomap,
-			NetClsClassid:     old.Resources.NetClsClassid,
-		},
-	}
-
-	result.Cgroups = newCgroup
-
-	return &result
-}
-
-func readState(dockerRoot, containerID string) (preAPIState, error) {
-	// pre-API libcontainer changed how its state was stored, try the old way of a "pid" file
-	statePath := configPath(dockerRoot, containerID)
-	if !utils.FileExists(statePath) {
-		pidPath := path.Join(dockerRoot, libcontainerExecDriverPath, containerID, "pid")
-		if utils.FileExists(pidPath) {
-			// We don't need the old state, return an empty state and we'll gracefully degrade.
-			return preAPIState{}, nil
-		}
-	}
-	out, err := ioutil.ReadFile(statePath)
-	if err != nil {
-		return preAPIState{}, err
-	}
-
-	// Parse the state.
-	var state preAPIState
-	err = json.Unmarshal(out, &state)
-	if err != nil {
-		return preAPIState{}, err
-	}
-
-	return state, nil
-}
-
 // Gets the path to the libcontainer configuration.
 func configPath(dockerRun, containerID string) string {
```

Comment:
It's used by the [`Exists()`](https://github.com/google/cadvisor/blob/a8edca0c4672434537e0c9f8b3fbe126fe7ed872/container/docker/handler.go#L401) function to determine whether the container still exists. We might be able to switch to the same method that raw & rkt use:
```go
func (self *rawContainerHandler) Exists() bool {
	// If any cgroup exists, the container is still alive.
	for _, cgroupPath := range self.cgroupPaths {
		if utils.FileExists(cgroupPath) {
			return true
		}
	}
	return false
}
```
But I'm not sure what the full implications of that are...